pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -38,29 +49,231 @@ type Service struct {
 	store             storage.Store
 }
 
+//state defines a didcomm service state
+type state struct {
+	MsgType string            `json:"@type"`
+	ID      string            `json:"@id"`
+	Thread  *decorator.Thread `json:"~thread"`
+	Current string
+}
+
+//key struct prepares the composite key for state persitance and lookup
+type key struct {
```

Comment:
this should include the thread ID.

It would be preferable if it also implements `String()`

Reply:
moved